tree/20_cfe_basics/log_rudder.cf

```diff
@@ -510,20 +510,25 @@ bundle agent _method_reporting_context(c_name, c_key)
 bundle agent _method_reporting_context_v4(c_name, c_key, report_id)
 {
   vars:
-    "report_data.component_name" string => "${c_name}";
-    "report_data.component_key"  string => "${c_key}";
-    # Non-canonified version, to be used in reports
-    "report_data.report_id_r"    string => "${report_id}";
-    # Canonified version, to be used in policies (as class prefix)
-    # it needs the directive_id to enforce unicity
-    "report_data.report_id"      string => canonify("${report_id}_${report_data.directive_id}");
-    "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}");
-    "report_data.identifier"     string => "${report_data.rule_id}@@${report_data.directive_id}@@${report_data.report_id_r}";
+    !pass1::
+      "report_data.index_eval" string => eval("${report_data.index}+1");
+      "report_data.index"      int => int("${report_data.index_eval}");
+      "report_data.component_name" string => "${c_name}";
+      "report_data.component_key"  string => "${c_key}";
+      # Non-canonified version, to be used in reports
+      "report_data.report_id_r"    string => "${report_id}";
+      # Canonified version, to be used in policies (as class prefix)
+      # it needs the directive_id to enforce unicity
+      "report_data.report_id"      string => canonify("${report_id}_${report_data.directive_id}");
+      "report_data.method_id"      string => canonify("${report_id}_${report_data.directive_id}_${report_data.index}");
```

Comment:
this change looks *really* complex.
we need to clearly define what is expected in report, method, and why index is necessary like (why not use something else like an auto generated id from the webapp) 
there are many moving parts, in services, packages in 20_services, but also in system techniques.

Reply:
replaced usage a method_id to report_id (they were the SAME value)

Reply:
To use report_id effectively in reports and method_id where we need to identify the method